decorator/base.rb

```diff
@@ -0,0 +1,12 @@
+require_relative '../nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
```

Comment:
- [ OPTIONAL ] Why don't you use the getter/setter method for the `nameable` instance variable so that it can be accessed? You can add `attr_accessor :nameable` to achieve that :blush: